elastic/kibana
# [Index Management] Allow to create a custom template from fleet with predefined values

## Description 

Related to https://github.com/elastic/kibana/issues/171540 
Similar to https://github.com/elastic/kibana/pull/134910

Context: We are making some change in Fleet to not create the custom component template by default, and allow the user to lazily create it.
 
To allow to create a template from fleet UI, we need to allow the create component template to have some default values and a redirect path 
 
<img width="400" alt="Screenshot 2024-01-23 at 3 39 06 PM" src="https://github.com/elastic/kibana/assets/1336873/492bdaef-9d0a-433d-9ded-608f92b4336f">


For this I made some change to make the create component template to accept some parameter (like we do for pipeline)
`name`, `managed_by`, `step`.

I also made some change to show the rollover modal as we did on edit

## How to test


Try to access this locally and see it work
http://localhost:5601/app/management/data/index_management/create_component_template/?name=logs-nginx.test@custom&step=mappings&redirect_path=/app/integrations/edit-integration/3a02668a-b1fb-4c2f-8067-6d64d0a45cd3?datastreamId=logfile-nginx.access-3a02668a-b1fb-4c2f-8067-6d64d0a45cd3


## Todo 

- [ ] Try to add some automated tests
